File: Leetcode/1431. Kids With the Greatest Number of Candies/solution.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	cur_max := max(candies)
	result := make([]bool, len(candies))

	for index, num := range candies {
		if num+extraCandies >= cur_max {
			result[index] = true
		} else {
			result[index] = false
		}
	}

	return result
}

func max(candies []int) int {
	max := 0

	for _, num := range candies {
		if num > max {
			max = num
		}
	}

	return max
}

File: Leetcode/1431. Kids With the Greatest Number of Candies/solution.rs
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {

        let mut result: Vec<bool> = Vec::new();
        let cur_max = Solution::max(&candies);

        for num in candies.iter() {
            if (num + extra_candies) >= cur_max{
                result.push(true);
            }
            else{
                result.push(false);
            }
        }
        result
    }
    
    pub fn max(list: &Vec<i32>) -> i32 {
        let mut cur_max: i32 = 0; // Nobody can have less than 1 candy
        for num in list.iter() {
            if num > &cur_max {
                cur_max = *num;
            }
        }
        cur_max
    }
}